src/node_buffer.cc

```diff
@@ -835,7 +835,9 @@ void Compare(const FunctionCallbackInfo<Value> &args) {
 
   size_t cmp_length = MIN(obj_a_length, obj_b_length);
 
-  int32_t val = memcmp(obj_a_data, obj_b_data, cmp_length);
+  int32_t val = 0;
```

Comment:
Fair enough. Just don't use a ternary.

Good call on the `int32_t`. Either just use an `int`, or `intptr_t`. The former is probably more straight forward.